lib/spack/spack/test/database.py

```diff
@@ -894,3 +894,36 @@ def _raise(db, spec):
     with pytest.raises(Exception):
         with spack.store.db.prefix_write_lock(s):
             assert False
+
+
+@pytest.mark.parametrize('spec_str,parent_name,expected_nparents', [
+    ('dyninst', 'callpath', 3),
+    ('libelf', 'dyninst', 1),
+    ('libelf', 'libdwarf', 1)
+])
+@pytest.mark.regression('11983')
+def test_check_parents(spec_str, parent_name, expected_nparents, database):
+    """Check that a spec returns the correct number of parents."""
+    s = database.query_one(spec_str)
+
+    parents = s.dependents(name=parent_name)
+    assert len(parents) == expected_nparents
+
+    edges = s.edges_from_dependents(name=parent_name)
+    assert len(edges) == expected_nparents
```

Comment:
No, it's really about how many parent specs this spec has. Our mock database looks like this:

https://github.com/spack/spack/blob/b7f4a421e4028115ba437fab7080c30e41d1b0d1/lib/spack/spack/test/conftest.py#L680-L699

so `dyninst` has 3 `callpath` parents. This just checks that we can retrieve all the three edges. Before this PR it was not possible since we couldn't distinguish among different `callpath` specs in `Spec._detendents`.

To be clear, the dependency type is just a edge property - so a `build,link` dependency from spec `foo/abcd` to spec `bar/1234` still counts a a single edge.